### DISCUSSION TITLE
Would need rollback of rdiff-backup 2.4.0 back to 2.2.6

### DISCUSSION THREAD

**Discussion by ericzolf** [Homebrew/discussions#4882] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

Hi,

after validation with @dtrodrigues I'm posting here to have our discussion in the open.

As the main developer of rdiff-backup, I erroneously released v2.4.0 which should never have gone out. Sadly it did, and you picked it before I noticed my error, meaning it's released now in Homebrew.

Could you please rollback your changes and bring back the version to v2.2.6? Here would be the commit to revert: https://github.com/Homebrew/homebrew-core/commit/fbee2f79750ee9898f51685a801814b81193fff7 as per https://github.com/Homebrew/homebrew-core/commits/master/Formula/r/rdiff-backup.rb

The most annoying thing is that Homebrew being 2.4.0 marks all other distros as being outdated [[1](https://github.com/rdiff-backup/rdiff-backup#5-packaging-status-in-distros)], which is quite annoying. And as I'm in the middle of a major rewrite, I can't release a higher version for the foreseeable future to "cover up" this one.

I'm absolutely sorry for the mess I've created and would appreciate your help in this matter.

Thanks, Eric

**Comment by motzke** [comment]
rdiff-backup v2.4.0 from homebrew doesn't seem to work, please roll back to v2.2.6. 
Until then you can run
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/4603ab2cbb2c9aee34de3beb06bc237cf5d19e18/Formula/r/rdiff-backup.rb

**Comment by samford** [comment, marked as answer]
@ericzolf I recently noticed the version mismatch in `rdiff-backup` and created a PR to revert it to 2.2.6 (https://github.com/Homebrew/homebrew-core/pull/156900). This surfaced issues in how we handle version downgrades in homebrew/core and I ended up having to modify our testing workflow to make certain types of version downgrades more feasible.

The aforementioned PR has been merged, so the `rdiff-backup` formula is back to version 2.2.6 (and Repology is correctly reporting it as newest again). **Any users on 2.4.0 will need to run `brew reinstall rdiff-backup` to downgrade to 2.2.6** (unlike casks, formula downgrades don't trigger a reinstall).

-----

If you release another 2.4.0 version in the future, anyone who updates the formula to that version will need to make sure it has a revision. We normally remove any `revision` call when upgrading to a newer version but we've already upgraded to 2.4.0 once before, so we will need `revision 1` for 2.4.0.

Without a revision, any users on the removed 2.4.0 version [who didn't see this and reinstall to downgrade] will remain there until the version exceeds 2.4.0. To be clear, this only applies to version 2.4.0 and everything will be back to normal with the next version higher than that (e.g., 2.4.1).

With that in mind, there may be something to be said for simply avoiding a subsequent 2.4.0 version in favor of releasing 2.4.1 instead when the time comes (i.e., this would be a typical version bump and a straightforward upgrade for existing users).

**Comment by ericzolf** [comment]
Thanks for the fix, and sorry again for the extra effort! According to plan, the next released version should be 3.0, so no issue. Kind regards!
